Core

kds1660 opened the pull request:
1) add new state - state menu between enterPin and EnterSum
2) add cash controller for adding nominals for cash view
3) add in cashModule method - return nominals in module for menu

kds1660 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into core

kds1660 committed:
chnge core and modules

1) add menu in at, after pin entered
2)add cash controller for adding nominals for cash view
3) add in cashModule method - return nominals in module for menu
4) add new state - state menu between enterPin and EnterSum

Varkh commented:
Nice

Varkh commented on js/controller/core.js:
```diff
@@ -49,6 +50,81 @@ function Core(cashModule, cardModule, navigation) {
+        stateMenu = new State("MENU", modules, null, {
+            cardPush: onCardPushHandler,
+            init: function () {
+                navigation.createMenu(['','','','View balanse',cashModule.getNominals()[0],cashModule.getNominals()[1],'Enter Cash','Back',],0)
```
I think it would be more readable in object form:
```
navigation.createMenu({
   4: 'View balanse',
   5: cashModule.getNominals()[0],
   6: cashModule.getNominals()[1],
   7: 'Enter Cash',
   8: 'Back'
})
```

You can create method that can handle both, array and object

Varkh commented on js/controller/core.js:
```diff
@@ -115,11 +191,62 @@ function Core(cashModule, cardModule, navigation) {
+                        setTimeout(function () {
+                            currectState = stateWait;
+                            currectState.init();
+                        }, 1000);
```
1) move all numbers to constants
2) Increase to 2500

Varkh reviewed (changes requested):
See comments

kds1660 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into core

kds1660 committed:
changes

made a request changes, time in variable, method createMenu accept array
or object

Varkh commented on js/view/displayView.js:
```diff
@@ -25,6 +25,41 @@ function DisplayView() {
+	this.createMenu=function (val,append) {
+		var table=$('<TABLE>');
+
+		for (var i=0;i<4;i++) {
```
This is jQuery, no need for `for`

kds1660 replied:
'for' for select value[i] element

kds1660 replied:
maybe understand, change

Varkh commented on js/view/displayView.js:
```diff
@@ -25,6 +25,41 @@ function DisplayView() {
+
+		for (var i=0;i<4;i++) {
+			var tr=($('<tr>'));
+			tr.css('height','30px');
```
Move to css

Varkh commented on js/view/displayView.js:
```diff
@@ -25,6 +25,41 @@ function DisplayView() {
+			tr.append($('<td>'));
+			table.append(tr)
+		}
+		table.css('width','100%');
```
To css

Varkh commented on js/view/displayView.js:
```diff
@@ -25,6 +25,41 @@ function DisplayView() {
+		table.css('width','100%');
+        table.css('margin-top','18px');
+
+		for (var i=0;i<4;i++) {
```
Use jQuery

Varkh commented on js/view/displayView.js:
```diff
@@ -25,6 +25,41 @@ function DisplayView() {
+				table.find('td:odd').eq(i).text(val[i+4]);
+			} else {
+				if (!val[i+1]) val[i+1]='';
+				table.find('td:even').eq(i).text(val[i+1]);
```
Use css

Varkh reviewed (commented):
Refactor
